components/constellation/constellation.rs

```diff
@@ -1161,9 +1161,7 @@ where
         {
             Some(id) => id.clone(),
             None => {
-                warn!(
-                    "Top-level was unpexpectedly removed from its top_level_browsing_context_set."
```

Comment:
So I think we should follow this rule:

Are we printing a complete sentence?

 - Yes: Use a capital letter and punctuation.
 - No: Don't use punctuation.

Following this rule, this statement need to maintain its current form. In addition, I think that messages that are for the user should err on the side of complete sentences as it is a part of the user interface. Trace and debugging output is for developers and it's less important that it be polished and professional, so can err on the side of terse sentence fragments.

So I guess here you could either make it a sentence fragment or add the punctuation.

Reply:
Maybe this could be:

```rust
    warn!("{top_level_id}: unexpectedly removed from its top_level_browsing_context_set");

Reply:
hmm, i’m not sure i agree. when reading a log message with only one sentence or sentence fragment, the full stop is implied by the end of the message, so adding a full stop conveys no additional meaning, and (ime) makes it harder to see exclamation marks or ellipses. that said, i agree it makes sense to use sentence case for messages without a prefix.

Reply:
I think of log statements as part of the UI, so it makes sense that we should maintain some professionalism and polish when writing them. In addition, you end up needing punctuation anyway when:

 1. You have more than one sentence in a log statement.
 2. You are logging a sentence with an exclamation mark or a question mark.
 
We should shoot for rules that are simple and consistent, which is why I think we should use punctuation versus having a rule that like "only use punctuation when logging a question." This just makes it harder to explain the rules to people contributing code.

Reply:
did the sentence case in 9772e8a36bd55a3619c20199e5081cfee8491cab per our conversation :)